File: demos/csharp-feequote/src/FeeQuote/Legacy/Fees.cs
// Retained for the nightly batch reconciliation job. Scheduled for removal once
// reconciliation moves to core pricing (see ticket 4821).

namespace FeeQuote.Legacy;

public static class Fees
{
    /// <summary>Applies a basis-point rate to an amount and returns whole cents.</summary>
    public static long CalculateFee(long amountCents, int rateBps)
    {
        return amountCents * rateBps / 10_000;
    }
}

File: demos/typescript-feequote/src/legacy/fees.ts
// Retained for the nightly batch reconciliation job. Scheduled for removal once
// reconciliation moves to core pricing (see ticket 4821).

const BPS_PER_WHOLE = 10000n;

/**
 * Fee for an amount at a rate expressed in basis points, in whole cents.
 */
export function calculateFee(amountCents: number, rateBps: number): number {
  return Number((BigInt(amountCents) * BigInt(rateBps)) / BPS_PER_WHOLE);
}
